com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CheckForOperationsInProgressStep.java

```diff
@@ -0,0 +1,128 @@
+package com.sap.cloud.lm.sl.cf.process.steps;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.cloudfoundry.client.lib.domain.CloudService;
+import org.flowable.engine.delegate.DelegateExecution;
+import org.springframework.stereotype.Component;
+
+import com.sap.cloud.lm.sl.cf.client.lib.domain.CloudServiceExtended;
+import com.sap.cloud.lm.sl.cf.client.lib.domain.ImmutableCloudServiceExtended;
+import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperation;
+import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperationState;
+import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperationType;
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
+import com.sap.cloud.lm.sl.cf.process.util.ServiceOperationGetter;
+import com.sap.cloud.lm.sl.cf.process.util.ServiceProgressReporter;
+import com.sap.cloud.lm.sl.common.util.JsonUtil;
+
+@Component("checkForOperationsInProgressStep")
+public class CheckForOperationsInProgressStep extends AsyncFlowableStep {
+
+    @Inject
+    private ServiceOperationGetter serviceOperationGetter;
+    @Inject
+    private ServiceProgressReporter serviceProgressReporter;
+
+    @Override
+    protected StepPhase executeAsyncStep(ExecutionWrapper execution) throws Exception {
+        List<CloudServiceExtended> servicesToProcess = getServicesToProcess(execution);
+
+        List<CloudServiceExtended> existingServices = getExistingServices(execution.getControllerClient(), servicesToProcess);
+        if (existingServices.isEmpty()) {
+            return StepPhase.DONE;
+        }
+
+        Map<CloudServiceExtended, ServiceOperation> servicesInProgressState = getServicesInProgressState(execution, existingServices);
+        if (servicesInProgressState.isEmpty()) {
+            return StepPhase.DONE;
+        }
+
+        getStepLogger().info(Messages.POLLING_IN_PROGRESS_SERVICES);
+
+        Map<String, ServiceOperationType> servicesOperationTypes = getServicesOperationTypes(servicesInProgressState);
+        getStepLogger().debug(Messages.SERVICES_IN_PROGRESS, JsonUtil.toJson(servicesOperationTypes, true));
+        StepsUtil.setTriggeredServiceOperations(execution.getContext(), servicesOperationTypes);
+
+        List<CloudServiceExtended> servicesData = getServicesData(servicesInProgressState);
+        StepsUtil.setServicesData(execution.getContext(), servicesData);
+
+        return StepPhase.POLL;
+    }
+
+    protected List<CloudServiceExtended> getServicesToProcess(ExecutionWrapper execution) {
+        return Collections.singletonList(StepsUtil.getServiceToProcess(execution.getContext()));
```

Comment:
In almost every place Arrays.asList() is used in this sort of context, even if only one element is needed.
Can we use that for consistency?